docs/ingress.md

```diff
@@ -0,0 +1,76 @@
+# Ingress
+
+Tyk Operator offers an Ingress Controller, which dynamically manages ApiDefinition resources as per the ingress spec.
+ Tyk Gateway can be configured as a drop-in replacement for any standard Kubernetes Ingress.
+
+Most Ingress Controllers heavily rely on annotations to configure the ingress gateway. With Tyk Operator, our Ingress 
+ Controller prefers to reference a strongly typed custom resource template.
+
+## Motivation
+
+The standard Ingress resource is very basic and does not natively support many advanced capabilities that are required
+ for real-life use-cases. Despite this, the community have built tooling, capabilities, dependencies, and technical debt 
+ on top of the ingress resource. These all rely on abusing the metadata annotations, and this practice is apparent 
+ with the standard [Kubernetes NginX Ingress resource](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#annotations).
+
+In order to decouple, gain all the benefits of Kubernetes, offer a Native & consistent API, we introduced the Tyk
+ ApiDefinition custom resource, which brings Full-Lifecycle API Management capabilities to Kubernetes. A side effect of
+ this meant a potential trade-off between offering a K8s native experience & integrating with Ingress which would 
+ facilitate clean integration with 3rd party tooling built on-top of and dependent on the Ingress Resource.
+
+As a compromise & attempt to propose an alternative & more scalable solution, we have introduced the concept of the 
+ ingress template ApiDefinition resource. The Template ApiDefinition resource offers a means to extend the capabilities 
+ of the standard Ingress Resource, by merging features of the ingress specification with that of the template.
+
+## Sample Template ApiDefinition resource
+
+```yaml
+apiVersion: tyk.tyk.io/v1alpha1
+kind: ApiDefinition
+metadata:
+  name: myapideftemplate
+  labels:
+    - isIngressTemplate: true
```

Comment:
I am having a second thought about this. Seeing that in our ingress we have annotation

```
tyk.io/template: myapideftemplate
```

Maybe we should have something  similar like 

```
  labels:
    - tyk.io/template: true
```

I have seen this pattern even on the k8s  api's (can't remember where) but keeping it consistent is a good think so we don't have to introduce too many labels. We will only have `tyk.io/template` , which acts as a filter on `APIDefinition` and used to annotate the ingress resource.

Reply:
Or go even further and use `tyk.io/template: myapideftemplate` for both ingress and template API def, and we matches them to link them together.

Reply:
```
apiVersion: tyk.tyk.io/v1alpha1
kind: ApiDefinition
```

But we are already inside a Tyk object. So we don't need to be careful about prefixing with `tyk.io`. Usually you add the prefix to resources that you don't own in order to play nice with other tooling.

e.g. on the Ingress resource: `cert-manager.io/issuer: my-issuer` because Ingress is not owned by cert-manager.

Reply:
If other tools want to extend / integrate with a Tyk ApiDefinition resource - they are welcome to prefix their annotations.

Reply:
That makes sense.

@asoorm what about 

```
  labels:
    - template: true
```

Reply:
It's not like we will be having different kinds of `APIDefinition` templates, this will only mark the Object as a template.